### ISSUE TITLE
Accessibility improvement on Calendar

### ISSUE THREAD

**Issue by sejui** [primefaces/primeng#4167]
### There is no guarantee in receiving a response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeNG PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

http://plnkr.co/edit/6q5v84DWolH6DeAdi70w?p=preview

**Current behavior**
Not really a bug, but the p-calendar's date picker does not meet standards for accessibility (WCAG/508 compliance). For the calendar ui button (the little calendar that you click to open the date picker), when the screen reader reads it, it only shows "ui button" which is not readable for people who cannot see. Same as "next month" and "previous month" buttons on the date picker, they only show "ui button" on the screen reader.

**Expected behavior**
add aria-label to these three buttons. Ex: aria-label = "date picker", aria-label = "previous month", aria-label = "next month"

**What is the motivation / use case for changing the behavior?**
the accessibility Law is coming from Ontario and all companies in Canada will need accessibility widgets. 

**Please tell us about your environment:**
Mac OS

* **Angular version:** 2.0.0-beta.13

* **PrimeNG version:** 2.0.1

* **Browser:** [all | Chrome | Firefox | IE 10,11+ | Safari 
 
* **Language:** TypeScript 2.3.4

* **Node (for AoT issues):** `node --version` =   5.4.1

**Comment by akashkriplani** [comment]
@cagataycivici is there any update on this enhancement?

**Comment by apaech** [comment]
> **Expected behavior**
> add aria-label to these three buttons. Ex: aria-label = "date picker", aria-label = "previous month", aria-label = "next month"

I suggest to depend second word on mode which is set by "view" attribute i.e. "next month" or "next year".
Also these labels should be translatable using https://primefaces.org/primeng/showcase/#/i18n

**Comment by blakejolley** [comment]
We've also discovered that this fails WCAG 4.1.2 and needed to fix the issue in our upcoming release

Whilst we wait for a proper fix, I've used the `MutationObserver` to attempt to alleviate the issue. 
Posting rough code here in case others need to 'patch' the calendars. Disclaimer: It's a bit brittle, could break in the future, but seems to work for our usecase. 

```
export class YourComponent implements OnInit, OnDestroy {

observer: MutationObserver;

  constructor(private elementRef: ElementRef) {
    this.observer = new MutationObserver(mutations => this.onMutate(mutations, this.elementRef.nativeElement));
    this.observer.observe(elementRef.nativeElement, {
      childList: true,
      subtree: true,
    });
  }

onMutate(mutations: MutationRecord[], componentNativeElement: HTMLElement) {
  // Get a list of headers for calendars that are still in the DOM (mutations can contain items that are removed by the time you inspect the DOM)
    const modifiedHeaders: HTMLElement[] = mutations
      .filter(mutation =>  mutation.target.nodeType === 1 && (mutation.target as HTMLElement).classList.contains('p-datepicker-header'))
      .map(mutation => mutation.target as HTMLElement)
      .filter(targetEl => componentNativeElement.contains(targetEl));

    for (const calendarHeader of modifiedHeaders) {
      this.tryToAddAriaLabelsToCalendarNavButtons(calendarHeader);
    }
  }

  updateAriaAttributesIfMissing(nativeElement, selector: string, label: string) {
    for (const element of [...nativeElement.querySelectorAll(selector)]) {
      const ariaLabel = element.getAttribute('aria-label');
      if (!ariaLabel || ariaLabel.length === 0) {
        element.setAttribute('aria-label', label);
      }
    }
  }

  tryToAddAriaLabelsToCalendarNavButtons(parentEl: HTMLElement) {
    this.updateAriaAttributesIfMissing(parentEl, '.p-datepicker-prev', 'Previous');
    this.updateAriaAttributesIfMissing(parentEl, '.p-datepicker-next', 'Next');
  }

ngOnDestroy() {
    this.observer.disconnect();
  }
}
```
